packages/three-vrm/src/VRMUtils/combineSkeletons.ts

```diff
@@ -222,3 +235,62 @@ function matrixEquals(a: THREE.Matrix4, b: THREE.Matrix4, tolerance?: number) {
 
   return true;
 }
+
+/**
+ * Check if the contents of two arrays are equal.
+ */
+function arrayEquals<T>(a: T[], b: T[]): boolean {
+  if (a.length !== b.length) {
+    return false;
+  }
+
+  return a.every((v, i) => v === b[i]);
+}
+
+/**
+ * Collect skin index attributes and corresponding bone arrays from the given skinned meshes.
+ * If a skin index attribute is shared among different bone sets, clone the attribute.
+ */
+function collectSkinIndexAttrs(
+  meshes: Iterable<THREE.SkinnedMesh>,
+): Set<[THREE.BufferAttribute | THREE.InterleavedBufferAttribute, THREE.Bone[]]> {
+  const skinIndexBonesPairSet = new Set<[THREE.BufferAttribute | THREE.InterleavedBufferAttribute, THREE.Bone[]]>();
+
+  // Collect skin index attributes
+  // skinIndex attribute might be shared among different bone sets
+  // If there are multiple bone sets that share the same skinIndex attribute, clone the attribute
+  const skinIndexNewSkinIndexBonesMapMap = new Map<
+    THREE.BufferAttribute | THREE.InterleavedBufferAttribute,
+    Map<THREE.BufferAttribute | THREE.InterleavedBufferAttribute, THREE.Bone[]>
+  >();
+
+  for (const mesh of meshes) {
+    let skinIndexAttr = mesh.geometry.getAttribute('skinIndex');
+
+    // Get or create a map for the skin index attribute
+    let newSkinIndexBonesMap = skinIndexNewSkinIndexBonesMapMap.get(skinIndexAttr);
+    if (newSkinIndexBonesMap == null) {
+      // Create a new map for the skin index attribute and register the bone array
+      newSkinIndexBonesMap = new Map();
+      skinIndexNewSkinIndexBonesMapMap.set(skinIndexAttr, newSkinIndexBonesMap);
+      newSkinIndexBonesMap.set(skinIndexAttr, mesh.skeleton.bones);
+      skinIndexBonesPairSet.add([skinIndexAttr, mesh.skeleton.bones]);
+      continue;
+    }
+
+    // Check if the bone set is already registered
+    for (const bones of newSkinIndexBonesMap.values()) {
+      if (arrayEquals(bones, mesh.skeleton.bones)) {
+        continue;
+      }
+    }
```

Comment:
There is a problem with the scope of the `continue`.

Reply:
oh no

Reply:
fixed @ https://github.com/pixiv/three-vrm/pull/1569/commits/902e6d83a4048a1e7a6bf4dfdb51252d7f8eb456

I also forgot to set the new attribute when we found an already registered skin index attribute 😅